### PR TITLE
Shorten execution time of HPA tests

### LINKED ISSUE #15732: Unit tests for pod autoscaler takes too long
Currently they take ~24 seconds. That's too long for unit tests. I've noticed some sleeps. We should get rid of them.

@quinton-hoole

### PR DESCRIPTION
**Pull request by mwielgus**
Fixes #15732. Reduces execution time to 4 sec.

### CODE REVIEW

**Comment by piosz** [comment]
LGTM

FYI @lavalamp @caesarxuchao

**Comment by eparis** [comment]
@k8s-bot test this, issue #22435

File: pkg/controller/podautoscaler/horizontal_test.go [hunk 1]
```diff
@@ -247,12 +249,23 @@ func (tc *testCase) verifyResults(t *testing.T) {
...
+		scaleNamespacer: testClient.Extensions(),
+		hpaNamespacer:   testClient.Extensions(),
+	}
+
 	err := hpaController.reconcileAutoscalers()
 	assert.Equal(t, nil, err)
 	if tc.verifyEvents {
 		// We need to wait for events to be broadcasted (sleep for longer than record.sleepDuration).
-		time.Sleep(12 * time.Second)
+		time.Sleep(2 * time.Second)
```
Comment (lavalamp) [inline pkg/controller/podautoscaler/horizontal_test.go]: I would prefer you do this in a loop or something, this looks like it will make it flaky?

Comment (piosz) [inline pkg/controller/podautoscaler/horizontal_test.go]: @mwielgus please fix

Comment (mwielgus) [inline pkg/controller/podautoscaler/horizontal_test.go]: I think that doing it super right would require broadcaster refactoring. I would rather not do. This change should not make this test any more flaky.

Comment (piosz) [inline pkg/controller/podautoscaler/horizontal_test.go]: I believe what @lavalamp means here is to check whether events were created in a loop with timeout.

Comment (mwielgus) [inline pkg/controller/podautoscaler/horizontal_test.go]: The problem is that sometimes we want an event to be NOT created. Removing initial sleep will cause false positives.

**Comment by piosz** [comment]
LGTM

File: pkg/controller/podautoscaler/horizontal_test.go (hunk 1 above)
Comment (piosz) [inline pkg/controller/podautoscaler/horizontal_test.go]: OK

Comment (lavalamp) [inline pkg/controller/podautoscaler/horizontal_test.go]: I still maintain that this is flaky, but I guess you're not making it worse.

**Comment by wojtek-t** [comment]
@mwielgus - please rebase

**Comment by piosz** [comment]
ping @mwielgus

**Comment by bgrant0607** [comment]
Is this needed for 1.2?

**Comment by mwielgus** [comment]
No.

**Comment by piosz** [comment]
ping @mwielgus

**Comment by mwielgus** [comment]
@piosz PTAL

**Comment by piosz** [comment]
LGTM

**Pull request merged**